### PR TITLE
Use struct for label config instead of pointer

### PR DESCRIPTION
**Pull request by matthyx**
ref #11517

### CODE REVIEW

File: prow/plugins/label/label.go [hunk 1]
```diff
@@ -42,19 +43,22 @@ func init() {
...
-	labels := []string{"kind", "priority", "area"}
-	if config.Label != nil {
-		labels = append(labels, config.Label.AdditionalLabels...)
-	}
+func configString(labels []string) string {
 	var formattedLabels []string
 	for _, label := range labels {
 		formattedLabels = append(formattedLabels, fmt.Sprintf(`"%s/*"`, label))
 	}
+	return fmt.Sprintf("The label plugin will work on %s and %s labels.", strings.Join(formattedLabels[:len(formattedLabels)-1], ", "), formattedLabels[len(formattedLabels)-1])
```
Comment (matthyx) [inline prow/plugins/label/label.go]: Actually found a bug here, as it was skipping the one before last label.

File: prow/plugins/config.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ type Configuration struct {
 	ConfigUpdater              ConfigUpdater          `json:"config_updater,omitempty"`
 	Golint                     Golint                 `json:"golint"`
 	Heart                      Heart                  `json:"heart,omitempty"`
-	Label                      *Label                 `json:"label,omitempty"`
+	Label                      Label                  `json:"label"`
```
Comment (matthyx) [inline prow/plugins/config.go]: Does it make sense to remove the `omitempty` here? I mean, if you don't specify it in json, you will get a default struct, won't you?

Comment (stevekuznetsov) [reply prow/plugins/config.go]: It is technically used also when writing out, if it's empty and `omitempty` it won't get written when you serialize a config. but I am not aware of anyone doing this? Could keep it to be safe? @cjwagner @fejta

**Comment by cjwagner** [comment]
I don't feel strongly either way about pointer+omitempty or non-pointer versions of structs for config. It is way nicer to read serialized config that uses pointers+omitempty, but it isn't a big deal.
So long as we don't need to distinguish a provided empty value from an unprovided value these changes should be fine.

**Comment by stevekuznetsov** [comment]
@cjwagner serialized config using `omitempty` without pointers also works really well -- you get nothing written if the zero value is in memory before writing

**Comment by stevekuznetsov** [comment]
We don't need to distinguish here.
/hold cancel

**Pull request merged**